modules/Microsoft.Compute/sshPublicKeys/.test/common/dependencies.bicep

```diff
@@ -0,0 +1,61 @@
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+@description('Optional. Name of the Deployment Script that creates the SSH Public Key.')
+param generateSshPubKeyScriptName string
+
+@description('Required. The name of the Managed Identity to create.')
+param managedIdentityName string
+
+@description('Required. Name of the temporary SSH Public Key to create for test.')
+param tempKeyName string
+
+@description('Optional. Do not provide a value. Used to force the deployment script to rerun on every redeployment.')
+param utcValue string = utcNow()
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+    name: managedIdentityName
+    location: location
+}
+
+// required for the deployment script to create a new temporary ssh public key object
+resource msi_ContributorRoleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+    name: guid(resourceGroup().id, 'ManagedIdentityContributor', '<<namePrefix>>')
+    properties: {
+        roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c') // Contributor
+        principalId: managedIdentity.properties.principalId
+        principalType: 'ServicePrincipal'
+    }
+}
+
+resource createPubKeyScript 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
+    name: generateSshPubKeyScriptName
+    location: location
+    kind: 'AzurePowerShell'
+    identity: {
+        type: 'UserAssigned'
+        userAssignedIdentities: {
+            '${managedIdentity.id}': {}
+        }
+    }
+    properties: {
+        azPowerShellVersion: '8.0'
+        retentionInterval: 'P1D'
+        arguments: '-resourceGroupName ${resourceGroup().name} -keyName ${tempKeyName}'
+        scriptContent: loadTextContent('../.scripts/New-SshPubKey.ps1')
+        cleanupPreference: 'OnExpiration'
+        forceUpdateTag: utcValue
```

Comment:
Is this needed? In our [Linux VM](https://github.com/Azure/ResourceModules/blob/main/modules/Microsoft.Compute/virtualMachines/.test/linux.min/dependencies.bicep#:~:text=%7D-,resource%20sshDeploymentScript%20%27Microsoft.Resources/deploymentScripts%402020%2D10%2D01%27%20%3D%20%7B,%7D,-resource%20sshKey%20%27Microsoft) implementation (that uses SSH keys) we didn't use the property. That naturally, if it makes sense to provide it, we can add it there.